cmd/namespace/delete.go

```diff
@@ -76,3 +90,77 @@ func (nc *NamespaceCommand) ExecuteDeleteNamespace(ctx context.Context, namespac
 	}
 	return nil
 }
+
+func (nc *NamespaceCommand) watchDelete(ctx context.Context, namespace string) error {
+	waitCtx, ctxCancel := context.WithCancel(ctx)
+	defer ctxCancel()
+
+	stop := make(chan os.Signal, 1)
+	defer close(stop)
+
+	signal.Notify(stop, os.Interrupt)
+	exit := make(chan error, 1)
+	defer close(exit)
+
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func(wg *sync.WaitGroup) {
+		defer wg.Done()
+		exit <- nc.waitForNamespaceDeleted(waitCtx, namespace)
+	}(&wg)
+
+	wg.Add(1)
+	go func(wg *sync.WaitGroup) {
+		defer wg.Done()
+		err := nc.okClient.Stream().DestroyAllLogs(waitCtx, namespace)
+		if err != nil {
+			oktetoLog.Warning("delete namespace logs cannot be streamed due to connectivity issues")
+			oktetoLog.Infof("delete namespace logs cannot be streamed due to connectivity issues: %v", err)
+		}
+	}(&wg)
+
+	select {
+	case <-stop:
+		ctxCancel()
+		oktetoLog.Infof("CTRL+C received, exit")
+		return oktetoErrors.ErrIntSig
+	case err := <-exit:
+		// wait until streaming logs have finished
+		wg.Wait()
+		return err
+	}
+}
+
+func (nc *NamespaceCommand) waitForNamespaceDeleted(ctx context.Context, namespace string) error {
+	timeout := 5 * time.Minute
+	ticker := time.NewTicker(1 * time.Second)
+	to := time.NewTicker(timeout)
+
+	for {
+		select {
+		case <-to.C:
+			return fmt.Errorf("%w: namespace %s, time %s", errDeleteNamespaceTimeout, namespace, timeout.String())
+		case <-ticker.C:
+			ns, err := nc.k8sClient.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
+			if err != nil {
+				// not found error is expected when the namespace is deleted.
+				// adding also IsForbidden as in some versions kubernetes returns this status when the namespace is deleted
+				// one or the other we assume the namespace has been deleted
+				if k8sErrors.IsNotFound(err) || k8sErrors.IsForbidden(err) {
+					return nil
+				}
+				return err
+			}
+
+			status, ok := ns.Labels["space.okteto.com/status"]
+			if !ok {
+				// when status label is not present, continue polling the namespace until timeout
```

Comment:
Sorry for the back and forth Tere, could we include a `oktetoLog.Debugf` to comment that the namespace doesn't have a label? In case we have reports about the command taking time, we could hve that log to see if it could be a problem

Reply:
Talked offline to modify also `waitForNamespaceDestroyAllToComplete` to behave in the same way (add a debug message and `continue`)